i18n(zh-tw): Create cloudflare.mdx (#9050)
withastro/docs: src/content/docs/zh-tw/guides/deploy/cloudflare.mdx

---
title: 部署你的 Astro 網站至 Cloudflare Pages
description: 如何使用 Cloudflare Pages 將你的 Astro 網站部署到網路上。
i18nReady: true
---
import ReadMore from '~/components/ReadMore.astro';
import { Steps } from '@astrojs/starlight/components';

> 你可以將你的Astro項目部署在 [Cloudflare Pages](https://pages.cloudflare.com/)，一個供前端開發人員協作和部署靜態（JAMstack）或 SSR 網站的平臺。

Reviewer: ```suggestion
你可以將你的 Astro 項目部署在 [Cloudflare Pages](https://pages.cloudflare.com/)，一個供前端開發人員協作和部署靜態（JAMstack）或 SSR 網站的平臺。
```
還有一點空格的小問題

本指南包含：

> - [如何使用 Git 部署一個站點](#如何使用-Git-部署一個站點)

Reviewer 2: 我看原文是 `How to deploy through the Cloudflare Pages Dashboard`，這邊的譯文是誤植了嗎？

> - [如何使用 Wrangler 部署一個站點](#如何使用-Wrangler-部署一個站點)
> - [如何部署 SSR 站點](#如何部署-SSR-站點)

Reviewer 2: 稍微調整得更符合臺灣用語～

```suggestion
- [如何使用 Wrangler 部署](#如何使用-Wrangler-部署)
- [如何部署 SSR 網站](#如何部署-SSR-網站)
```

## 前提條件

開始之前，你需要：

> - 一個 Cloudflare 賬戶，如果你沒有已經擁有一個，你可以創建一個免費的 Cloudflare 賬戶在過程中

Reviewer 2: 這句的語法感覺不那麼像中文，我調整了一下前後順序和一些用詞

```suggestion
- 一個 Cloudflare 帳號，如果還沒有的話，可以在過程中建立免費的 Cloudflare 帳號
```

> - 你的代碼存儲在一個 [Github](https://github.com/) 或者 [Gitlab](https://about.gitlab.com/) 賬戶裡

Reviewer 2: ```suggestion
- 把程式碼推到 [Github](https://github.com/) 或者 [GitLab](https://about.gitlab.com/) 儲存庫
```

> ## 如何使用 Git 部署一個站點

Reviewer 2: ```suggestion
## 如何使用 Git 部署網站
```

<Steps>

> 1. 在 Cloudflare Pages 創立一個新項目

Reviewer 2: ```suggestion
1. 在 Cloudflare Pages 建立新專案
```

> 2. 將你的代碼提交到一個 Git 倉庫中（Github，Gitlab）

Reviewer 2: ```suggestion
2. 將程式碼推到 git 倉儲（GitHub、GitLab）
```

> 3. 登錄 Cloudflare Dashboard 並在 **Account Home** > **Workers & Pages** > **Overview** 中選擇你的賬戶

Reviewer 2: ```suggestion
3. 登錄 Cloudflare Dashboard 並在 **Account Home** > **Workers & Pages** > **Overview** 中選擇你的帳號
```

> 4. 選擇 **Create application**，然後選擇 **Pages** 標籤頁，接著選擇 **Connect to Git** 選項。

Reviewer 2: ```suggestion
4. 選擇 **Create application**，接著選 **Pages** 標籤，最後選 **Connect to Git** 選項。
```

稍微調整了一下，減少同一個詞「選擇」重複出現的狀況

> 5. 選擇你想部署的 Git 專案並點選 **Begin setup**（初始設定）。

Reviewer 2: ```suggestion
5. 選擇你想部署的 git 專案並點選 **Begin setup**（初始設定）。
```

> 6. 使用以下的構建設定：
> 
>     - **Framework preset（框架預設）:** `Astro`
>     - **Build command（構建命令）:** `npm run build`
>     - **Build output directory（構建輸出目錄）:** `dist`

Reviewer 2: ```suggestion
6. 使用以下的建構設定：

    - **Framework preset:** `Astro`
    - **Build command:** `npm run build`
    - **Build output directory:** `dist`
```

專有的文字我傾向統一維持原文～

> 7. 點選 **Save and Deploy**（儲存並部署）按鈕。

Reviewer 2: ```suggestion
7. 點選 **Save and Deploy** 按鈕。
```

專有的文字我傾向統一維持原文～

</Steps>

>  如何使用 Wrangler 部署一個站點

Reviewer 2: ```suggestion
## 如何使用 Wrangler 部署網站
```

<Steps>

> 1. 安裝 [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/)。
> 
> 2. 使用 `wrangler login` 在 Wrangler 登陸 Cloudflare 賬號並授權。
> 
> 3. 執行你的構建命令（比如 `npm run build`）。
> 
> 4. 使用 `npx wrangler pages deploy dist` 進行部署。

Reviewer 2: ```suggestion
1. 安裝 [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/)。

2. 使用 `wrangler login` 登錄 Cloudflare 帳號並授權 Wrangler。

3. 執行你的建構命令。

4. 使用 `npx wrangler pages deploy dist` 部署。
```

</Steps>

```bash

> # 安裝 Wrangler CLI（命令列）
> npm install -g wrangler
> # 在 CLI 中登陸 Cloudflare 賬號
> wrangler login
> # 執行你的構建命令
> npm run build
> # 建立新的部署
> npx wrangler pages deploy dist

Reviewer 2: ```suggestion
# 安裝 Wrangler CLI（命令列）
npm install -g wrangler
# 透過 CLI 登入 Cloudflare 帳號
wrangler login
# 執行你的建構命令
npm run build
# 建立新的部署
npx wrangler pages deploy dist
```

> 上傳完所有的檔案後，Wrangler 將為你提供一個預覽 URL 以檢查你的站點。當你登入 Cloudflare Pages 儀表板時，你將看到你的新專案。

Reviewer 2: ```suggestion
上傳完所有檔案後，Wrangler 會提供一個預覽網址讓你檢查網站。登入 Cloudflare Pages 儀表板時，你就會看到新專案。
```

### 使用 Wrangler 在本地啟用預覽

> 要使預覽版正常工作，你必須安裝 `wrangler`

Reviewer 2: ```suggestion
要使預覽正常運作，你必須安裝 `wrangler`
```

```bash
pnpm add wrangler --save-dev
```

然後就可以將預覽指令碼從 Astro 的內建預覽命令更新為 `wrangler` 了：

```json title="package.json"
"preview": "wrangler pages dev ./dist"
```

> ## 如何部署 SSR 站點

Reviewer 2: ```suggestion
## 如何部署 SSR 網站
```

> 你可以使用 [`@astrojs/cloudflare` 介面卡](/zh-tw/guides/integrations-guide/cloudflare/)將 Astro SSR 站點部署到 Cloudflare Pages。

Reviewer 2: ```suggestion
你可以使用 [`@astrojs/cloudflare` 轉接器](/zh-tw/guides/integrations-guide/cloudflare/)將 Astro SSR 站點部署到 Cloudflare Pages。
```

> 請按照以下步驟設定介面卡。完成後，你可以使用上述文件中的任何方法進行部署。

Reviewer 2: ```suggestion
請按照以下步驟設定轉接器。完成後，你可以使用上述文件中的任何方法進行部署。
```

### 快速安裝

> 使用以下 `astro add` 命令新增 Cloudflare 介面卡以在你的 Astro 專案中啟用 SSR。這將安裝介面卡並一步對你的檔案 `astro.config.mjs` 進行適當的更改。

Reviewer 2: ```suggestion
使用以下 `astro add` 命令新增 Cloudflare 轉接器以在你的 Astro 專案中啟用 SSR。這將安裝轉接器並一併對 `astro.config.mjs` 進行適當的更改。
```

```bash
npx astro add cloudflare
```

### 手動安裝

> 如果你想要手動安裝介面卡，請完成以下兩個步驟：
> 
> <Steps>
> 1. 使用你喜歡的包管理器將 `@astrojs/cloudflare` 新增到專案的依賴項中。如果你正在使用 npm 或不確定是哪個包管理器，請在終端中執行：
> 
>     ```bash
>     npm install @astrojs/cloudflare
>     ```
> 
> 2. 將以下內容新增到你的 `astro.config.mjs` 檔案中：
> 
>     ```js title="astro.config.mjs" ins={2, 5-6}
>     import { defineConfig } from 'astro/config';
>     import cloudflare from '@astrojs/cloudflare';
> 
>     export default defineConfig({
>       output: 'server',
>       adapter: cloudflare()
>     });
>     ```
> </Steps>
> 
> <ReadMore>閱讀更多關於 [Astro 中的 SSR（服務端渲染）](/zh-tw/guides/server-side-rendering/) 的資訊。</ReadMore>

Reviewer 2: ```suggestion
如果你想要手動安裝轉接器，請完成以下兩個步驟：

<Steps>
1. 使用你喜歡的套件管理器將 `@astrojs/cloudflare` 加到專案的相依套件中。如果你使用 npm 或不確定是哪個管理器，請在終端機執行：

    ```bash
    npm install @astrojs/cloudflare
    ```

2. 將以下內容新增到 `astro.config.mjs` 檔案中：

    ```js title="astro.config.mjs" ins={2, 5-6}
    import { defineConfig } from 'astro/config';
    import cloudflare from '@astrojs/cloudflare';

    export default defineConfig({
      output: 'server',
      adapter: cloudflare()
    });
    ```
</Steps>

<ReadMore>進一步了解 [Astro 的 SSR（伺服器端算繪）](/zh-tw/guides/server-side-rendering/)。</ReadMore>
```

## 故障排除

### 客戶端 hydration

> 由於 Cloudflare 的 Auto Minify（自動壓縮）功能，客戶端 Hydration 可能會失敗。如果你在控制檯中看到 `Hydration completed but contains mismatches`，請確保在 Cloudflare 設定中禁用 Auto Minify。

Reviewer 2: ```suggestion
Cloudflare 的 Auto Minify（自動壓縮）功能可能導致客戶端 hydration 失敗。如果你在控制檯中看到 `Hydration completed but contains mismatches`，請確保在 Cloudflare 設定中停用 Auto Minify。
```

> ### Node.js 執行時 API
> 
> 如果你正在構建一個使用 [Cloudflare 服務端渲染介面卡](/zh-tw/guides/integrations-guide/cloudflare/) 的按需渲染專案，並且伺服器在構建時失敗，出現如 `[Error] Could not resolve "XXXX. The package "XXXX" wasn't found on the file system but is built into node.` 的錯誤資訊：
> 
> - 這意味著你在伺服器端環境中使用的某個包或匯入項與 [Cloudflare 執行時 API](https://developers.cloudflare.com/workers/runtime-apis/nodejs/) 不相容。
> 
> - 如果你直接匯入了一個 Node.js 執行時 API，請參考 Astro 關於 Cloudflare 的 [Node.js 相容性](/zh-cn/guides/integrations-guide/cloudflare/#nodejs-相容性) 文件，瞭解如何解決這個問題的進一步步驟。
> 
> - 如果你匯入的包中包含了 Node.js 執行時 API，請檢查該包的作者是否支援 `node:*` 的匯入語法。如果不支援，你可能需要找到一個替代的包。

Reviewer 2: ```suggestion
### Node.js runtime API

如果你用 [Cloudflare SSR 轉接器](/zh-tw/guides/integrations-guide/cloudflare/) 的隨需算繪建立專案，而伺服器在建構時失敗，出現如 `[Error] Could not resolve "XXXX. The package "XXXX" wasn't found on the file system but is built into node.` 的錯誤資訊：

- 這意味著你在伺服器端環境中使用的某個套件或匯入與 [Cloudflare runtime API](https://developers.cloudflare.com/workers/runtime-apis/nodejs/) 不相容。

- 如果你直接匯入 Node.js runtime API，請參考 Astro 關於 Cloudflare 的 [Node.js 相容性](/zh-cn/guides/integrations-guide/cloudflare/#nodejs-相容性)文件，瞭解如何進一步解決這個問題。

- 如果你匯入的套件中包含 Node.js runtime API，請檢查該套件的作者是否支援 `node:*` 的匯入語法。如果不支援，你可能需要找到其他替代套件。
```

---

> title: 部署你的 Astro 網站至 Cloudflare Pages
> description: 如何使用 Cloudflare Pages 將你的 Astro 網站部署到網路上。
> i18nReady: true

Reviewer 3: ```suggestion
title: 部署你的 Astro 網站至 Cloudflare Pages
description: 如何使用 Cloudflare Pages 將你的 Astro 網站部署到網路上。
type: deploy
i18nReady: true
```

这行 `type: deploy` 目的是用来表示属于 deploy（部署你的网站）文档的，不需要删除。

---
import ReadMore from '~/components/ReadMore.astro';
import { Steps } from '@astrojs/starlight/components';

你可以將你的Astro項目部署在 [Cloudflare Pages](https://pages.cloudflare.com/)，一個供前端開發人員協作和部署靜態（JAMstack）或 SSR 網站的平臺。

本指南包含：

- [如何使用 Git 部署一個站點](#如何使用-Git-部署一個站點)
- [如何使用 Wrangler 部署一個站點](#如何使用-Wrangler-部署一個站點)
- [如何部署 SSR 站點](#如何部署-SSR-站點)

## 前提條件

開始之前，你需要：

> - 一個Cloudflare賬戶，如果你沒有已經擁有一個，你可以創建一個免費的Cloudflare賬戶在過程中
> - 你的代碼存儲在一個[Github](https://github.com/)或者[Gitlab](https://about.gitlab.com/)賬戶裡

Reviewer 3: ```suggestion
- 一個 Cloudflare 賬戶，如果你沒有已經擁有一個，你可以創建一個免費的 Cloudflare 賬戶在過程中
- 你的代碼存儲在一個 [Github](https://github.com/) 或者 [Gitlab](https://about.gitlab.com/) 賬戶裡
```

## 如何使用 Git 部署一個站點

<Steps>
1. 在 Cloudflare Pages 創立一個新項目

2. 將你的代碼提交到一個 Git 倉庫中（Github，Gitlab）

3. 登錄 Cloudflare Dashboard 並在 **Account Home** > **Workers & Pages** > **Overview** 中選擇你的賬戶

4. 選擇 **Create application**，然後選擇 **Pages** 標籤頁，接著選擇 **Connect to Git** 選項。

5. 選擇你想部署的 Git 專案並點選 **Begin setup**（初始設定）。

6. 使用以下的構建設定：

    - **Framework preset（框架預設）:** `Astro`
    - **Build command（構建命令）:** `npm run build`
    - **Build output directory（構建輸出目錄）:** `dist`

7. 點選 **Save and Deploy**（儲存並部署）按鈕。
</Steps>

 如何使用 Wrangler 部署一個站點

<Steps>
1. 安裝 [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/)。

2. 使用 `wrangler login` 在 Wrangler 登陸 Cloudflare 賬號並授權。

3. 執行你的構建命令（比如 `npm run build`）。

4. 使用 `npx wrangler pages deploy dist` 進行部署。
</Steps>

> ```bash
>  安裝 Wrangler CLI（命令列）
> npm install -g wrangler
>  在 CLI 中登陸 Cloudflare 賬號
> wrangler login
>  執行你的構建命令
> npm run build
>  建立新的部署
> npx wrangler pages deploy dist
> ```

Reviewer 3: ```suggestion
```bash
# 安裝 Wrangler CLI（命令列）
npm install -g wrangler
# 在 CLI 中登陸 Cloudflare 賬號
wrangler login
# 執行你的構建命令
npm run build
# 建立新的部署
npx wrangler pages deploy dist
```
```
这里的 `#` 是 命令行的注释语法，最后会被渲染成注释。

---

> title: 部署你的Astro網站至Cloudflare Pages

Reviewer 3: ```suggestion
title: 部署你的 Astro 網站至 Cloudflare Pages
```

description: 如何使用 Cloudflare Pages 將你的 Astro 網站部署到網路上。
type: deploy
i18nReady: true
---
import ReadMore from '~/components/ReadMore.astro';
import { Steps } from '@astrojs/starlight/components';

> 你可以將你的Astro項目部署在[Cloudflare Pages](https://pages.cloudflare.com/)，一個平臺供前端開發人員協作和部署靜態（JAMstack）或SSR網站的平臺

Reviewer 3: ```suggestion
你可以將你的Astro項目部署在 [Cloudflare Pages](https://pages.cloudflare.com/)，一個供前端開發人員協作和部署靜態（JAMstack）或 SSR 網站的平臺。
```

本指南包含：

> - [如何使用Git部署一個站點](#如何使用Git部署一個站點)
> - [如何使用Wrangler部署一個站點](#如何使用Wrangler部署一個站點)
> - [如何部署SSR站點](#如何部署SSR站點)

Reviewer 3: ```suggestion
- [如何使用 Git 部署一個站點](#如何使用-Git-部署一個站點)
- [如何使用 Wrangler 部署一個站點](#如何使用-Wrangler-部署一個站點)
- [如何部署 SSR 站點](#如何部署-SSR-站點)
```

## 前提條件

開始之前，你需要：

- 一個Cloudflare賬戶，如果你沒有已經擁有一個，你可以創建一個免費的Cloudflare賬戶在過程中
- 你的代碼存儲在一個[Github](https://github.com/)或者[Gitlab](https://about.gitlab.com/)賬戶裡

> ## 如何使用Git部署一個站點

Reviewer 3: ```suggestion
## 如何使用 Git 部署一個站點
```

> <steps>

Reviewer 3: ```suggestion
<Steps>
```

日志中的报告描述了是这个组件的 `大小写问题`，导致了 `CI预览部署的失败`。

> 1.在Cloudflare Pages創立一個新項目
> 
> 2.將你的代碼提交到一個 Git 倉庫中（Github，Gitlab）
> 
> 3.登錄Cloudflare Dashboard並在 **Account Home** > **Workers & Pages** > **Overview** 中選擇你的賬戶

Reviewer 3: ```suggestion
1. 在 Cloudflare Pages 創立一個新項目

2. 將你的代碼提交到一個 Git 倉庫中（Github，Gitlab）

3. 登錄 Cloudflare Dashboard 並在 **Account Home** > **Workers & Pages** > **Overview** 中選擇你的賬戶
```

序数后面需要一个空格，来保证步骤被解析成 `<Step>` 组件的一部分。

4. 選擇 **Create application**，然後選擇 **Pages** 標籤頁，接著選擇 **Connect to Git** 選項。

5. 選擇你想部署的 Git 專案並點選 **Begin setup**（初始設定）。

6. 使用以下的構建設定：

    - **Framework preset（框架預設）:** `Astro`
    - **Build command（構建命令）:** `npm run build`
    - **Build output directory（構建輸出目錄）:** `dist`

7. 點選 **Save and Deploy**（儲存並部署）按鈕。
</Steps>

## 如何使用 Wrangler 部署一個站點

<Steps>
1. 安裝 [Wrangler CLI](https://developers.cloudflare.com/workers/wrangler/get-started/)。

2. 使用 `wrangler login` 在 Wrangler 登陸 Cloudflare 賬號並授權。

3. 執行你的構建命令（比如 `npm run build`）。

4. 使用 `npx wrangler pages deploy dist` 進行部署。
</Steps>

```bash
# 安裝 Wrangler CLI（命令列）
npm install -g wrangler
# 在 CLI 中登陸 Cloudflare 賬號
wrangler login
# 執行你的構建命令
npm run build
# 建立新的部署
npx wrangler pages deploy dist
```

上傳完所有的檔案後，Wrangler 將為你提供一個預覽 URL 以檢查你的站點。當你登入 Cloudflare Pages 儀表板時，你將看到你的新專案。

### 使用 Wrangler 在本地啟用預覽

要使預覽版正常工作，你必須安裝 `wrangler`

```bash
pnpm add wrangler --save-dev
```

然後就可以將預覽指令碼從 Astro 的內建預覽命令更新為 `wrangler` 了：

```json title="package.json"
"preview": "wrangler pages dev ./dist"
```

## 如何部署 SSR 站點

> 你可以使用 [`@astrojs/cloudflare` 介面卡](/zh-cn/guides/integrations-guide/cloudflare/)將 Astro SSR 站點部署到 Cloudflare Pages。

Reviewer 3: ```suggestion
你可以使用 [`@astrojs/cloudflare` 介面卡](/zh-tw/guides/integrations-guide/cloudflare/)將 Astro SSR 站點部署到 Cloudflare Pages。
```

請按照以下步驟設定介面卡。完成後，你可以使用上述文件中的任何方法進行部署。

### 快速安裝

使用以下 `astro add` 命令新增 Cloudflare 介面卡以在你的 Astro 專案中啟用 SSR。這將安裝介面卡並一步對你的檔案 `astro.config.mjs` 進行適當的更改。

```bash
npx astro add cloudflare
```

### 手動安裝

如果你想要手動安裝介面卡，請完成以下兩個步驟：

<Steps>
1. 使用你喜歡的包管理器將 `@astrojs/cloudflare` 新增到專案的依賴項中。如果你正在使用 npm 或不確定是哪個包管理器，請在終端中執行：

    ```bash
    npm install @astrojs/cloudflare
    ```

2. 將以下內容新增到你的 `astro.config.mjs` 檔案中：

    ```js title="astro.config.mjs" ins={2, 5-6}
    import { defineConfig } from 'astro/config';
    import cloudflare from '@astrojs/cloudflare';

    export default defineConfig({
      output: 'server',
      adapter: cloudflare()
    });
    ```
</Steps>

> <ReadMore>閱讀更多關於 [Astro 中的 SSR（服務端渲染）](/zh-cn/guides/server-side-rendering/) 的資訊。</ReadMore>

Reviewer 3: ```suggestion
<ReadMore>閱讀更多關於 [Astro 中的 SSR（服務端渲染）](/zh-tw/guides/server-side-rendering/) 的資訊。</ReadMore>
```

## 故障排除

### 客戶端 hydration

由於 Cloudflare 的 Auto Minify（自動壓縮）功能，客戶端 Hydration 可能會失敗。如果你在控制檯中看到 `Hydration completed but contains mismatches`，請確保在 Cloudflare 設定中禁用 Auto Minify。

### Node.js 執行時 API

> 如果你正在構建一個使用 [Cloudflare 服務端渲染介面卡](/zh-cn/guides/integrations-guide/cloudflare/) 的按需渲染專案，並且伺服器在構建時失敗，出現如 `[Error] Could not resolve "XXXX. The package "XXXX" wasn't found on the file system but is built into node.` 的錯誤資訊：

Reviewer 3: ```suggestion
如果你正在構建一個使用 [Cloudflare 服務端渲染介面卡](/zh-tw/guides/integrations-guide/cloudflare/) 的按需渲染專案，並且伺服器在構建時失敗，出現如 `[Error] Could not resolve "XXXX. The package "XXXX" wasn't found on the file system but is built into node.` 的錯誤資訊：
```
